native/cocos/platform/android/java/vendor/google/billing/GoogleBillingHelper.java

```diff
@@ -24,191 +24,512 @@ of this software and associated documentation files (the "Software"), to deal
 
+import com.android.billingclient.api.AcknowledgePurchaseResponseListener;
+import com.android.billingclient.api.AlternativeBillingOnlyAvailabilityListener;
+import com.android.billingclient.api.AlternativeBillingOnlyInformationDialogListener;
 import com.android.billingclient.api.AlternativeBillingOnlyReportingDetails;
+import com.android.billingclient.api.AlternativeBillingOnlyReportingDetailsListener;
+import com.android.billingclient.api.BillingClient;
 import com.android.billingclient.api.BillingConfig;
+import com.android.billingclient.api.BillingConfigResponseListener;
+import com.android.billingclient.api.BillingFlowParams;
 import com.android.billingclient.api.BillingResult;
+import com.android.billingclient.api.ConsumeResponseListener;
+import com.android.billingclient.api.ExternalOfferAvailabilityListener;
+import com.android.billingclient.api.ExternalOfferInformationDialogListener;
 import com.android.billingclient.api.ExternalOfferReportingDetails;
+import com.android.billingclient.api.ExternalOfferReportingDetailsListener;
+import com.android.billingclient.api.InAppMessageParams;
+import com.android.billingclient.api.InAppMessageResponseListener;
 import com.android.billingclient.api.InAppMessageResult;
 import com.android.billingclient.api.ProductDetails;
+import com.android.billingclient.api.ProductDetailsResponseListener;
 import com.android.billingclient.api.Purchase;
 import java.util.List;
+
+import com.android.billingclient.api.PurchasesResponseListener;
+import com.android.billingclient.api.PurchasesUpdatedListener;
+import com.android.billingclient.api.UserChoiceBillingListener;
+import com.android.billingclient.api.UserChoiceDetails;
+import com.cocos.lib.CocosHelper;
 import com.cocos.lib.GlobalObject;
-import google.billing.GoogleBilling;
+import android.util.SparseArray;
+import com.android.billingclient.api.BillingClientStateListener;
 
 public class GoogleBillingHelper {
-    public static native void onBillingSetupFinished(@NonNull BillingResult billingResult);
-    public static native void onBillingServiceDisconnected();
-    public static native void onProductDetailsResponse(@NonNull BillingResult billingResult, @NonNull List<ProductDetails> productDetailsList, int startID);
-    public static native void onPurchasesUpdated(@NonNull BillingResult billingResult, @Nullable List<Purchase> purchasesList, int startID);
-    public static native void onQueryPurchasesResponse(@NonNull BillingResult billingResult, @NonNull List<Purchase> purchasesList, int startID);
-    public static native void onConsumeResponse(@NonNull BillingResult billingResult, @NonNull String purchaseToken);
-    public static native void onAcknowledgePurchaseResponse(@NonNull BillingResult billingResult);
-    public static native void onBillingConfigResponse(@NonNull BillingResult billingResult, @Nullable BillingConfig billingConfig);
-
-    public static native void onAlternativeBillingOnlyTokenResponse(@NonNull BillingResult billingResult,
+
```

Comment:
If there are no subclasses, it is recommended to declare it as a final class.

Reply:
fixed.